django/test/testcases.py

```diff
@@ -524,17 +524,26 @@ def assertContains(
             response, text, status_code, msg_prefix, html
         )
 
+        content = (
+            b"".join(response.streaming_content)
+            if response.streaming
+            else response.content
+        )
+
         if count is not None:
             self.assertEqual(
                 real_count,
                 count,
                 msg_prefix
-                + "Found %d instances of %s in response (expected %d)"
-                % (real_count, text_repr, count),
+                + "Found %d instances of %s (expected %d) in the following response\n%s"
+                % (real_count, text_repr, count, content),
             )
         else:
             self.assertTrue(
-                real_count != 0, msg_prefix + "Couldn't find %s in response" % text_repr
+                real_count != 0,
+                msg_prefix
+                + "Couldn't find %s in the following response\n%s"
+                % (text_repr, content),
```

Comment:
IMO, we should use `safe_repr(content, True)` in all methods.

Reply:
`safe_repr(..., short=True)` truncates a content so we should use `safe_repr(content)`. It also shows that we need better tests.